rq/worker.py

```diff
@@ -883,13 +900,18 @@ def handle_job_failure(self, job, queue, started_job_registry=None,
                     job_class=self.job_class
                 )
             job.worker_name = None
+
+            # check whether a job was stopped intentionally and set the job
+            # status appropriately if it was this job.
+            if self._stopped_job_id and self._stopped_job_id == job.id:
+                job.set_status(JobStatus.STOPPED)
+                self._stopped_job_id = None
+
             # Requeue/reschedule if retry is configured
-            if job.retries_left and job.retries_left > 0:
-                retry = True
-                retry_interval = job.get_retry_interval()
-                job.retries_left = job.retries_left - 1
-            else:
-                retry = False
+            # Don't attempt to retry if the job was deliberately stopped
+            retry = not job.is_stopped and job.retries_left and job.retries_left > 0
+            # Don't overwrite the JobStatus of stopped jobs
+            if not retry and not job.is_stopped:
                 job.set_status(JobStatus.FAILED, pipeline=pipeline)
```

Comment:
I think we should group the status setting logic together so it's easier to follow. Something like:
```python
if self._stopped_job_id and self._stopped_job_id == job.id:
    job.set_status(JobStatus.STOPPED, pipeline=pipeline)
    self._stopped_job_id = None
else:
    job.set_status(JobStatus.FAILED, pipeline=pipeline)
```

Reply:
Adjusted